shared/api/url.go

```diff
@@ -0,0 +1,30 @@
+package api
+
+import (
+	"net/url"
+	"strings"
+)
+
+// URLBuild helps to safely generate a URL to a LXD API endpoint.
+// It appends each of the pathParts (escaped using url.PathEscape) prefixed with "/" to the URL path.
+// The projectName will be added as an escaped query parameter "project" if it is non-empty or not "default".
+func URLBuild(projectName string, pathParts ...string) *url.URL {
```

Comment:
Should we also handle `target` while we're at it?

Reply:
Yes that would make sense.

Reply:
I've reworked this a little to add a new `api.URL` type that embeds `net.URL` and allows building up a URL safely using chained functions.